File: 169-MajorityElement/169-MajorityElement.cpp
// Last updated: 10/27/2025, 7:08:59 PM
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int freq=0,i,ans;
        for(i=0;i<n;i++){
            if(freq==0){
                ans=nums[i];
            }
            if(nums[i]==ans){

                freq++;
            }
            else{
                freq--;
            }
        }
        //for variation problem
        int count=0;
        for(int val:nums){
            if(val==ans){
                count++;
            }
        }
        if(count>n/2){
        return ans;
        }
        return -1;
    }
};

File: 169-MajorityElement/169-MajorityElement.py
# Last updated: 10/27/2025, 7:08:58 PM
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq=1
        ans=nums[0]
        for i in range(1,len(nums)):
            if ans==nums[i]:
                freq+=1
            else:
                freq-=1
            
            if freq==0:
                ans=nums[i]
                freq=1
        return ans

